### PR TITLE
[concurrency] SILGen: emit @asyncHandler functions.

### PR DESCRIPTION
**Pull request by eeckstein**
An asyncHandler function is split into two functions:
1. The asyncHandler body function: it contains the body of the function, but is emitted as an async function.
2. The original function: it just contains
      _runAsyncHandler(operation: asyncHandlerBodyFunction)

To distinguish the names of the original asyncHandler function and it's generated "body-function", the body-function name is mangled with an async attribute, i.e. as if it was declared as async.

rdar://problem/71247879

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci smoke test

File: lib/AST/ASTMangler.cpp [hunk 1]
```diff
@@ -2870,7 +2874,7 @@ void ASTMangler::appendEntity(const ValueDecl *decl, StringRef EntityOp,
     appendOperator("Z");
 }
 
-void ASTMangler::appendEntity(const ValueDecl *decl) {
+void ASTMangler::appendEntity(const ValueDecl *decl, bool isAsyncHandlerBody) {
```
Comment (DougGregor) [inline lib/AST/ASTMangler.cpp:2877]: Should this `bool` be a `FunctionManglingKind` instead to make it clearer?

**Review by DougGregor — APPROVED** [review]
Looking pretty good, thanks!

File: lib/AST/ASTMangler.cpp (hunk 1 above)
Comment (eeckstein) [reply lib/AST/ASTMangler.cpp:2877]: I thought of that, but at the end decided to pass a "bool", because FunctionManglingKind has 3 cases.

**Comment by eeckstein** [comment]
@swift-ci smoke test

**Pull request merged**